File: RepoTransBench/target_projects/Java/C#/awslabs_route53-infima/public_tests/TwoDimensionalLatticePublicTest.cs
using System;
using System.Collections.Generic;
using Xunit;

namespace Route53Infima.PublicTests
{
    public class TwoDimensionalLatticePublicTest
    {
        [Fact]
        public void TestAddAndGetEndpointsDifferentData()
        {
            var lattice = new TwoDimensionalLattice<string>("rack", "slot");
            lattice.AddEndpoints("rack-17", "slot-9", new List<string> { "R", "S", "T" });
            var found = lattice.GetEndpoints("rack-17", "slot-9");
            Assert.Contains("R", found);
            Assert.Contains("S", found);
            Assert.Contains("T", found);
        }

        [Fact]
        public void TestAddEndpointDifferentData()
        {
            var lattice = new TwoDimensionalLattice<string>("zoneA", "rowB");
            lattice.AddEndpoint("foo", "bar", "X");
            var found = lattice.GetEndpoints("foo", "bar");
            Assert.Contains("X", found);
            Assert.Single(found);
        }

        [Fact]
        public void TestGetEndpointsForNonexistentCoordinatesPublic()
        {
            var lattice = new TwoDimensionalLattice<string>("c", "d");
            Assert.Empty(lattice.GetEndpoints("doesnot", "exist"));
        }

        [Fact]
        public void TestDimensionNamesPublic()
        {
            var lattice = new TwoDimensionalLattice<string>("fruit", "color");
            Assert.Equal(new List<string> { "fruit", "color" }, lattice.GetDimensionNames());
        }
    }
}

File: RepoTransBench/source_projects/Java/awslabs_route53-infima/src/test/java/com/amazonaws/services/route53/infima/TwoDimensionalLatticePublicTest.java
package com.amazonaws.services.route53.infima;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.Arrays;
import java.util.Collection;

public class TwoDimensionalLatticePublicTest {
    @Test
    public void testAddAndGetEndpointsDifferentData() {
        TwoDimensionalLattice<String> lattice = new TwoDimensionalLattice<>("rack", "slot");
        lattice.addEndpoints("rack-17", "slot-9", Arrays.asList("R", "S", "T"));
        Collection<String> found = lattice.getEndpoints("rack-17", "slot-9");
        assertTrue(found.contains("R"));
        assertTrue(found.contains("S"));
        assertTrue(found.contains("T"));
    }

    @Test
    public void testAddEndpointDifferentData() {
        TwoDimensionalLattice<String> lattice = new TwoDimensionalLattice<>("zoneA", "rowB");
        lattice.addEndpoint("foo", "bar", "X");
        Collection<String> found = lattice.getEndpoints("foo", "bar");
        assertTrue(found.contains("X"));
        assertEquals(1, found.size());
    }

    @Test
    public void testGetEndpointsForNonexistentCoordinatesPublic() {
        TwoDimensionalLattice<String> lattice = new TwoDimensionalLattice<>("c", "d");
        assertTrue(lattice.getEndpoints("doesnot", "exist").isEmpty());
    }

    @Test
    public void testDimensionNamesPublic() {
        TwoDimensionalLattice<String> lattice = new TwoDimensionalLattice<>("fruit", "color");
        assertEquals(Arrays.asList("fruit", "color"), lattice.getDimensionNames());
    }
}